### PR TITLE
Add code examples in timeseries module's docstrings

### PR DESCRIPTION
**Pull request by datacubeR**
Adding code examples for:

- LagFeatures
- WindowFeatures
- ExpandingWindowFeatures

### CODE REVIEW

File: feature_engine/timeseries/forecasting/lag_features.py [hunk 1]
```diff
@@ -95,6 +95,27 @@ class LagFeatures(BaseForecastTransformer):
...
+    >>>                     x1 = [1,2,3,4,5],
+    >>>                     x2 = [6,7,8,9,10]
+    >>>                     ))
+    >>> lf = LagFeatures(periods=[1,2])
+    >>> lf.fit_transform(X)
+            date  x1  x2  x1_lag_1  x2_lag_1  x1_lag_2  x2_lag_2
+    0  2022-09-18   1   4       NaN       NaN       NaN       NaN
+    1  2022-10-27   2   5       1.0       4.0       NaN       NaN
+    2  2022-12-24   3   6       2.0       5.0       1.0       4.0
+    3  2023-01-09   4   7       3.0       6.0       2.0       5.0
```
Comment (solegalli) [inline feature_engine/timeseries/forecasting/lag_features.py:118]: Is the output correct? where is the 4 in x2_lag_2 coming from? there is no 4 in the original variable.

Also, could we make the timestamps consecutive? each one a different  but consecutive day? it's more representative of forecasting data like sales or stocks

File: feature_engine/timeseries/forecasting/expanding_window_features.py [hunk 2]
```diff
@@ -117,6 +117,28 @@ class ExpandingWindowFeatures(BaseForecastTransformer):
...
     pandas.aggregate
     pandas.shift
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.timeseries.forecasting import ExpandingWindowFeatures
+    >>> X = pd.DataFrame(dict(date = ["2022-09-18",
+    >>>                             "2022-10-27",
```
Comment (solegalli) [inline feature_engine/timeseries/forecasting/expanding_window_features.py:127]: picky I know: could we make the indentation nicer? I think the "2022- etc should be aligned with the one on the first line.

File: feature_engine/timeseries/forecasting/lag_features.py [hunk 3]
```diff
@@ -95,6 +95,27 @@ class LagFeatures(BaseForecastTransformer):
...
     --------
     pandas.shift
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.timeseries.forecasting import LagFeatures
+    >>> X = pd.DataFrame(dict(date = ["2022-09-18",
+    >>>                             "2022-10-27",
```
Comment (solegalli) [inline feature_engine/timeseries/forecasting/lag_features.py:105]: indent

File: feature_engine/timeseries/forecasting/window_features.py [hunk 4]
```diff
@@ -121,6 +121,28 @@ class WindowFeatures(BaseForecastTransformer):
...
     pandas.aggregate
     pandas.shift
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> from feature_engine.timeseries.forecasting import WindowFeatures
+    >>> X = pd.DataFrame(dict(date = ["2022-09-18",
+    >>>                             "2022-10-27",
```
Comment (solegalli) [inline feature_engine/timeseries/forecasting/window_features.py:131]: indent

File: feature_engine/timeseries/forecasting/window_features.py [hunk 5]
```diff
@@ -121,6 +121,28 @@ class WindowFeatures(BaseForecastTransformer):
...
+    >>>                             "2022-12-24",
+    >>>                             "2023-01-09",
+    >>>                             "2023-02-12"],
+    >>>                     x1 = [1,2,3,4,5],
+    >>>                     x2 = [6,7,8,9,10]
+    >>>                     ))
+    >>> wf = WindowFeatures(window = 2)
+    >>> lf.fit_transform(X)
+              date  x1  x2  x1_lag_1  x2_lag_1  x1_lag_2  x2_lag_2
+    0  2022-09-18   1   6       NaN       NaN       NaN       NaN
```
Comment (solegalli) [inline feature_engine/timeseries/forecasting/window_features.py:141]: I think the output is wrong. It should say window or something different in the name of the new features. Also I think the values might be wrong. Could you double check?

Also, could you make the timestamps consecutive in the dataframe?

**Review by solegalli — COMMENTED** [review]
Hey @datacubeR 

Thank you so much for the changes. And apologies for the delayed response.

The examples look good, but I think there are some mismatches between what is shown and what the real output should be. Could you please double check?

I added some comments below. Thank you!

File: feature_engine/timeseries/forecasting/lag_features.py (hunk 1 above)
Comment (datacubeR) [reply feature_engine/timeseries/forecasting/lag_features.py:118]: It is correct. I'm asking for lags 1 and 2. So the 4 is the first row, from column x2 shifted two periods. 
Fixing dates.

File: feature_engine/timeseries/forecasting/expanding_window_features.py (hunk 2 above)
Comment (datacubeR) [reply feature_engine/timeseries/forecasting/expanding_window_features.py:127]: hahaha. My editor thinks that is the correct indentation, that is why I left it that way. But sure, I can fix it. No worries.

**Pull request merged**